Fix the wrong link

Gsantomaggio opened the pull request:
The link `Writing and Running Integration Tests` on the [documentation](https://book.kubebuilder.io/reference/writing-tests.html) is wrong.

This fix links the URL to `../../../testing/integration.md` that seems to be the correct link!

Gsantomaggio commented:
/assign @pwittrock

Gsantomaggio commented:
/retest

camilamacedo86 commented:
/assing camilamacedo86

camilamacedo86 commented:
/ok-to-test

Gsantomaggio commented:
thank you  @camilamacedo86 
I am trying to understand why the build failed, not sure it is related to my PR.

camilamacedo86 commented:
Hi @Gsantomaggio, 

The build did not fail. What is failing are the tests in the prow, `pull-kubebuilder-test` and `pull-kubebuilder-e2e `. They have been in a developing process so no worry 👍 . You can click on in the Details to see what they call and check more info in the CONTRIBUTION.md doc

mengqiy commented:
I tried to netlify preview. It seem it doesn't work well. Are there too many `..`?

mengqiy commented:
`testing/integration.md` need to be moved to somewhere under `docs/book/src` to be picked up by the book.

Gsantomaggio commented:
hi @camilamacedo86 
I moved the directory and added the [end-to-end link](https://github.com/kubernetes-sigs/kubebuilder/pull/1208/files#diff-aa18f5a97cdab2913b34408705a78547R76)

It should be ok now.

mengqiy commented:
/retest

mengqiy commented:
If you check the netlify deploy-preview, it still gives a 404.
I suspect you may need to add entries in https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/SUMMARY.md to make it work.

Gsantomaggio commented:
@mengqiy @camilamacedo86 Thank you for your patience, I didn't know `netlify`.
( I learnt something new ) 
I checked the deploy, and now it is ok.

Gsantomaggio commented:
hi @camilamacedo86 @mengqiy 
is there something I can do to help to merge this PR?

There is someone that met the same issue:
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/kubebuilder/lY6HDuLkcEg/37Eq91njAAAJ

:) 

Thank you!

mengqiy commented on docs/book/src/reference/writing-tests.md:
```diff
@@ -3,7 +3,7 @@
 files generated for you by kubebuilder are fairly minimal.
 
-[Writing and Running Integration Tests](/reference/testing/integration.md) documents steps to consider when writing integration steps for your controllers, and available options for configuring your test control plane using [`envtest`](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/envtest).
+[Writing and Running Integration Tests](testing/integration.md) documents steps to consider when writing integration steps for your controllers, and available options for configuring your test control plane using [`envtest`](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/envtest).
```
> testing/integration.md

After taking a closer look at `docs/testing/integration.md` and `docs/testing/e2e.md`, I just realized that they are for kubebuilder v1 and are out-of-date.
You will need to do the following:
- Change the link to `/reference/testing/envtest.md`. (Please ensure it works in netlify preview)
- Please add a note on top of `docs/testing/integration.md` and `docs/testing/e2e.md`. e.g. `**This document is for kubebuilder v1 only**`
- Revert other changes in `SUMMARY.md` and `integration.md`.

Sorry for the confusion.
I will add a small doc for e2e testing afterwards.

DirectXMan12 replied:
IIRC, we intentionally left this out of the book since they were v1 articles.  We need to add v2 equivalents, but they aren't there yet.

Gsantomaggio replied:
Thank you both, I understood. It wasn't clear to me.
Going to fix.

draghuram commented:
BTW, I also tried to fix the link in #1233 (along with a typo).

Gsantomaggio commented:
Tried on [netlify](https://deploy-preview-1208--kubebuilder.netlify.com/reference/writing-tests.html)
Messages added.
Thank you for the patience.

camilamacedo86 commented:
Hi @Gsantomaggio, 

> @mengqiy @camilamacedo86 Thank you for your patience, I didn't know netlify.

The netlifly in a CI job. See: 

<img width="893" alt="Screenshot 2019-12-10 at 00 53 28" src="https://user-images.githubusercontent.com/7708031/70485706-93f07b80-1ae7-11ea-87af-5017e634f1fd.png">
 
Just click on in the `Details` to check the changes in the docs.

camilamacedo86 commented on docs/testing/e2e.md:
```diff
@@ -1,5 +1,7 @@
 **Running End-to-end Tests on Remote Clusters**
 
+**This document is for kubebuilder v1 only**
```
I do not think that we should add this comment. 
if It has something that changed then, since the V1 is deprecated, IHMO we should update the doc to show how to do the same in the V2 instead of adding this note.

Gsantomaggio replied:
I did what @mengqiy asked :) https://github.com/kubernetes-sigs/kubebuilder/pull/1208#discussion_r353362278

I can revert it, let me know

camilamacedo86 replied:
Oh I see. So all is fine 👍

Gsantomaggio commented:
>The netlifly in a CI job. See:

Thank you @camilamacedo86 
Please see https://github.com/kubernetes-sigs/kubebuilder/pull/1208#issuecomment-562100337 I alredy found it and tested :)!

camilamacedo86 commented:
/restest

mengqiy commented:
/lgtm
/approve
